ga4gh/datamodel/variants.py

```diff
@@ -410,6 +445,26 @@ def convertVariant(self, record, callSetIds):
         variant.id = self.getVariantId(variant)
         return variant
 
+    def getVariant(self, compoundId):
+        if compoundId.referenceName in self._chromFileMap:
+            varFileName = self._chromFileMap[compoundId.referenceName]
+        else:
+            raise exceptions.ObjectNotFoundException(compoundId)
+        start = int(compoundId.start)
+        # fetch start is exclusive, not inclusive
+        referenceName, startPosition, endPosition = \
+            self.sanitizeVariantFileFetch(
+                compoundId.referenceName, start, None)
+        cursor = self.getFileHandle(varFileName).fetch(
+            referenceName, startPosition, endPosition)
+        for record in cursor:
+            if start != record.start:
```

Comment:
What about
```python
variant = self.convertVariant(record, [])
if record.start == start and compoundId.md5 == self.hashVariant(variant):
    return variant
elif record.start > start:
    raise ObjectNotFoundException()
```
Most of the time, we'll hit the variant first go and so there's no point in trying to save a few cycles doing the convert variant process. This way we have only one method to get the hash. We do need to guard against the variant not existing, otherwise we could iterate over millions of variants before exiting the loop. Another way to do this I suppose is to put in an end coodinate. Is there any reason we can't use ``start + 1`` there?